src/lib/storage/dictionary_segment.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "abstract_segment.hpp"
+#include "value_segment.hpp"
+
+namespace opossum {
+
+class AbstractAttributeVector;
+
+// Dictionary is a specific segment type that stores all its values in a vector
```

Comment:
Not your fault, but sentences should end in full stop:
```suggestion
// Dictionary is a specific segment type that stores all its values in a vector.
```